client/src/sim.rs

```diff
@@ -162,14 +182,31 @@ impl Sim {
                         },
                     }
                 }
+
+                if let Some(params) = self.params.as_ref() {
+                    if let Some(my_position) = my_position {
+                        let graph = &self.graph;
+                        self.prediction.reconcile(
+                            msg.latest_input,
+                            my_position,
+                            |position, input| {
+                                CharacterControllerPass {
+                                    position,
+                                    input,
+                                    graph,
+                                    config: &params.sim_config,
+                                    dt_seconds: 1.0 / params.sim_config.rate as f32,
+                                }
+                                .step()
+                            },
+                        );
+                    }
+                }
```

Comment:
Why was this inlined out of `update_position`?

Reply:
The purpose of the `update_position` method was a bit unclear to me. It had the dual purpose of setting the position component of all relevant entities as well as updating the player entity's predictions. These purposes meshed less well together when velocity was introduced, as velocity affects position but doesn't live in the position component. I inlined it out to separate out this second purpose with the idea that it probably belongs in a different method.

I never moved it to a different method because my concern about the ad-hoc nature of passing component data around meant that it would add to the complexity of this PR, and I wasn't certain I was going the right direction.

I made a mistake when splitting my changes into two commits, since this change makes less sense outside this context.